modules/c/nrt/source/Utils.c

```diff
@@ -468,126 +468,163 @@ NRTAPI(char) nrt_Utils_cornersTypeAsCoordRep(nrt_CornersType type)
     return cornerRep;
 }
 
-static void normalize_dms(int* pDegrees, int* pMinutes, int* pSeconds)
+static int is_valid_dms(int degrees, int minutes, double seconds)
 {
-    /* Ensure seconds and minutes are still within valid range. */
-    *pMinutes += (*pSeconds / 60);
-    *pSeconds %= 60;
-
-    *pDegrees += (*pMinutes / 60);
-    *pMinutes %= 60;
-
-    *pDegrees %= 360;
-
-    const int minutes_to_seconds = 60;
-    const int degrees_to_seconds = 60 * minutes_to_seconds;
-    int total_seconds = (*pDegrees * degrees_to_seconds) + (*pMinutes * minutes_to_seconds) + *pSeconds;
-
-    *pDegrees = total_seconds / degrees_to_seconds;
-    total_seconds -= (*pDegrees * degrees_to_seconds);
-
-    *pMinutes = total_seconds / minutes_to_seconds;
-    total_seconds -= (*pMinutes * minutes_to_seconds);
-
-    *pSeconds = total_seconds;
-}
+    // these are maximums; values for lat/lon may be smaller
+    if (abs(degrees) >= 360)
+    {
+        return 0;
+    }
+    if (abs(minutes) >= 60)
+    {
+        return 0;
+    }
+    if (fabs(seconds) >= 60.0)
+    {
+        return 0;
+    }
 
-static void adjust_dms(int* pDegrees, int* pMinutes, int* pSeconds, char* pDir,
-    char positive_dir, char negative_dir)
-{
-    *pDir = positive_dir;
-    if (*pDegrees < 0)
+    // A negative value is allowed for the first non-zero value
+    if (degrees != 0)
     {
-        *pDir = negative_dir;
-        *pDegrees *= -1;
+        if ((minutes < 0) || (seconds < 0.0))
+        {
+            return 0;
+        }
     }
-    if (*pMinutes < 0)
+    else if (minutes != 0)
     {
-        *pDir = negative_dir;
-        *pMinutes *= -1;
+        if (seconds < 0.0)
+        {
+            return 0;
+        }
     }
-    if (*pSeconds < 0)
+
+    return 1;
+}
+
+static int is_valid_lat_dms(int degrees, int minutes, double seconds)
+{
+    if (abs(degrees) > 90)
     {
-        *pDir = negative_dir;
-        *pSeconds *= -1;
+        return 0;
     }
+    return is_valid_dms(degrees, minutes, seconds);
 }
 
 NRTPROT(void) nrt_Utils_geographicLatToCharArray(int degrees, int minutes,
-                                                 double seconds_, char *buffer7)
+                                                 double seconds, char *buffer7)
 {
-    int seconds = (int) round(seconds_);
+    assert(buffer7 != NULL);
```

Comment:
If buffer7 is NULL, can we just early return? That would give us the least 'explosive' result, and place the responsibility for any problems with this function back on the caller

Reply:
Passing a `NULL` buffer is an error and indicates a bug in the calling code.  There is no way to indicate this particular "error" result except to crash.  The `assert()` will just trigger the error a bit earlier and more obviously in debug builds.